stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -2491,3 +2491,93 @@ for (tritype, comptritype) in ((:LowerTriangular, :UpperTriangular),
     @eval /(u::TransposeAbsVec, A::$tritype{<:Any,<:Adjoint}) = transpose($comptritype(conj(parent(parent(A)))) \ u.parent)
     @eval /(u::TransposeAbsVec, A::$tritype{<:Any,<:Transpose}) = transpose(transpose(A) \ u.parent)
 end
+
+# Cube root of a 2x2 real-valued matrix with complex conjugate eigenvalues and equal diagonal values.
+# Reference [1]: Smith, M. I. (2003). A Schur Algorithm for Computing Matrix pth Roots.
+#   SIAM Journal on Matrix Analysis and Applications (Vol. 24, Issue 4, pp. 971–989).
+#   https://doi.org/10.1137/s0895479801392697
+function _cbrt_2x2!(A::AbstractMatrix{T}) where {T<:Real}
+    m, n = size(A)
+    (m == 2 && n == 2) || throw(ArgumentError("_cbrt_2x2!: Matrix A must be 2x2."))
+    (A[1,1] == A[2,2]) || throw(ArgumentError("_cbrt_2x2!: Matrix A must have equal diagonal values."))
+    (A[1,2]*A[2,1] < 0) || throw(ArgumentError("_cbrt_2x2!: Matrix A must have complex conjugate eigenvalues."))
+    μ = sqrt(-4*A[1,2]*A[2,1])/2
+    r = cbrt(sqrt(A[1,1]^2 + μ^2))
+    θ = atan(μ, A[1,1])
+    α = r*cos(θ/3)
+    β′ = r*sin(θ/3)/µ
+    A[1,1] = α
+    A[2,2] = α
+    A[1,2] = β′*A[1,2]
+    A[2,1] = β′*A[2,1]
+    return A
+end
+
+# Cube root of a block diagonal matrix with 1x1 and 2x2 blocks (output of Schur decomposition)
+function _cbrt_blkdiag_1x1_2x2!(A::AbstractMatrix{T}) where {T<:Real}
+    m, n = size(A)
+    (m == n) || throw(ArgumentError("_cbrt_blkdiag_1x1_2x2!: Matrix A must be square."))
+    # 2x2 and 1x1 blocks
+    I₂ = findall(x -> !iszero(x), diag(A,-1))
+    I₁ = setdiff(1:n, vcat(I₂, I₂.+1))
+    for i in I₂ @views _cbrt_2x2!(A[i:i+1,i:i+1]) end
+    for i in I₁ @views A[i,i] = cbrt(A[i,i]) end
+    return A, I₂, I₁
+end
+
+# Cube root of a quasi upper triangular matrix (output of Schur decomposition)
+# Reference [1]: Smith, M. I. (2003). A Schur Algorithm for Computing Matrix pth Roots.
+#   SIAM Journal on Matrix Analysis and Applications (Vol. 24, Issue 4, pp. 971–989).
+#   https://doi.org/10.1137/s0895479801392697
+function _cbrt_quasi_triu!(A::AbstractMatrix{T}) where {T<:Real}
+    m, n = size(A)
+    (m == n) || throw(ArgumentError("_cbrt_quasi_triu!: Matrix A must be square."))
+    A, I₂, I₁ = _cbrt_blkdiag_1x1_2x2!(A)
+    sizes = [if i ∈ I₁ 1 elseif i ∈ I₂ 2 else 0 end for i=1:n]
+    Σ = cumsum(sizes)
+    # Algorithm 4.3 in Reference [1]
+    Δ = I(4)
+    M_S₁ = zeros(T,2,2)
+    M_S₂ = zeros(T,2,2)
+    M_L₀ = zeros(T,4,4)
+    M_L₁ = zeros(T,4,4)
+    M_Bᵢⱼ⁽⁰⁾ = zeros(T,2,2)
+    M_Bᵢⱼ⁽¹⁾ = zeros(T,2,2)
+    for k = 1:n-1
+        for i = 1:n-k
+            if sizes[i] == 0 || sizes[i+k] == 0 continue end
+            k₁, k₂ = i+1+(sizes[i+1]==0), i+Σ[i+k-1]-Σ[i+1]+sizes[i+1]+(sizes[i+1]==0)
+            i₁, i₂, j₁, j₂, s₁, s₂ = i, i+sizes[i]-1, i+k, i+k+sizes[i+k]-1, sizes[i], sizes[i+k]
+            @views S₁ = reshape(M_S₁, s₁, :)[:, 1:s₁]
```

Comment:
Rather than having `@views` on every line`, it's cleaner to just put `@views` on the whole block (e.g. the whole `for` loop, or even the whole function).

Reply:
This is done now.